### PR TITLE
New KiwiKeebs Macro V2 (QMK/VIA) - And Update to KiwiKeebs Macro V1

### PR DESCRIPTION
**Pull request by AKiwi92**
Hi guys! I have updated the macro v1 to be via compatible.

I have also added in the new macro v2 which used different pinouts etc and is a totally new board.

Excited for this one, thanks again for all your help last time and hopefully i haven't done anything too wrong!

Ash

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [X] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [X] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [X] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [X] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [X] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by drashna** [comment]
This has some merge conflicts that need to be fixed. 

Rebasing should fix this issue. 
https://docs.qmk.fm/#/newbs_git_resolving_merge_conflicts?id=rebasing-your-changes

**Comment by AKiwi92** [comment]
Right so added readme files which the linter picked up as an issue and how its giving me something else which I really aren't sure about! Any advice is greatly appreciated thanks guys!

File: keyboards/kiwikeebs/macro-v2/config.h [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#pragma once
+
+#include "config_common.h"
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID    0x4b51
```
Comment (drashna) [inline keyboards/kiwikeebs/macro-v2/config.h:23]:
```suggestion
#define VENDOR_ID    0x4B51
```

File: keyboards/kiwikeebs/macro-v2/readme.md [hunk 2]
```diff
@@ -0,0 +1,19 @@
+# KiwiKeebs MacroBoard V1.0
+
+![KiwiKeebs](https://i.imgur.com/h5i9jBl.png)
```
Comment (drashna) [inline keyboards/kiwikeebs/macro-v2/readme.md:3]: This image is not valid, "has been removed".

File: keyboards/kiwikeebs/macro-v2/readme.md [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+A 6 Key PCB with Rotary Encoder Knob + Button made from solid CNC aluminium.
+
+* Keyboard Maintainer: [AKiwi92](https://github.com/akiwi92)
+* Hardware Supported: KiwiKeebs MacroBoard V1.0
+* Hardware Availability: [reddit.com](https://www.reddit.com/r/mechmarket/comments/ibijrd/ic_kiwikeebs_6key_macroboard_rotary_encoder_diy/)
+
+Make example for this keyboard (after setting up your build environment):
```
Comment (drashna) [inline keyboards/kiwikeebs/macro-v2/readme.md:11]: Could you add instructions on how to jump to the bootloader?

File: keyboards/kiwikeebs/macro-v2/readme.md [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+
+    make kiwikeebsmacro:default
```
Comment (drashna) [inline keyboards/kiwikeebs/macro-v2/readme.md:13]: This command isn't accurate, could you update it?

File: keyboards/kiwikeebs/macro/config.h [hunk 5]
```diff
@@ -20,7 +20,7 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
 #include "config_common.h"
 
 /* USB Device descriptor parameter */
-#define VENDOR_ID    0xFEED
+#define VENDOR_ID    0x4b51
```
Comment (drashna) [inline keyboards/kiwikeebs/macro/config.h:23]:
```suggestion
#define VENDOR_ID    0x4B51
```

File: keyboards/kiwikeebs/macro/info.json [hunk 6]
```diff
@@ -1,5 +1,4 @@
-{
-    "keyboard_name": "KiwiKeebs MacroBoard v.10",
+    "keyboard_name": "KiwiKeebs MacroBoard v1.0",
```
Comment (drashna) [inline keyboards/kiwikeebs/macro/info.json:1]:
```suggestion
{
    "keyboard_name": "KiwiKeebs MacroBoard v1.0",
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/kiwikeebs/macro-v2/readme.md (hunk 2 above)
Comment (AKiwi92) [reply keyboards/kiwikeebs/macro-v2/readme.md:3]: Added with updated static logo on website

File: keyboards/kiwikeebs/macro-v2/readme.md (hunk 3 above)
Comment (AKiwi92) [reply keyboards/kiwikeebs/macro-v2/readme.md:11]: Added instructions on how to put into DFU mode

File: keyboards/kiwikeebs/macro-v2/readme.md (hunk 4 above)
Comment (AKiwi92) [reply keyboards/kiwikeebs/macro-v2/readme.md:13]: updated to add -v2

**Comment by AKiwi92** [comment]
I *think* I have done everything but let me know if you spot something else! :)
Thanks again!

File: keyboards/kiwikeebs/macro/readme.md [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+
+A 6 Key PCB with Rotary Encoder Knob + Button made from solid CNC aluminium.
+
+* Keyboard Maintainer: [AKiwi92](https://github.com/akiwi92)
+* Hardware Supported: KiwiKeebs MacroBoard V1.0
+* Hardware Availability: [reddit.com](https://www.reddit.com/r/mechmarket/comments/ibijrd/ic_kiwikeebs_6key_macroboard_rotary_encoder_diy/)
+
+Make example for this keyboard (after setting up your build environment):
+
+    make kiwikeebsmacro:default
```
Comment (drashna) [inline keyboards/kiwikeebs/macro/readme.md:13]:
```suggestion
    make kiwikeebs/macro:default
```

**Comment by AKiwi92** [comment]
Thanks again for the help, I think I have fixed everything this time :D

**Review by drashna — APPROVED** [review]

**Review by fauxpark — CHANGES_REQUESTED** [review]
Files with `-` and `.` in the name need to be changed to `_`s.

**Comment by AKiwi92** [comment]
Let me know if I have messed anything else up guys :D

File: keyboards/kiwikeebs/macro_v2/readme.md [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+* Hardware Supported: KiwiKeebs MacroBoard V1.0
+* Hardware Availability: [reddit.com](https://www.reddit.com/r/mechmarket/comments/ibijrd/ic_kiwikeebs_6key_macroboard_rotary_encoder_diy/)
+
+Make example for this keyboard (after setting up your build environment):
+
+    make kiwikeebsmacro:default
+
+Flashing example for this keyboard:
+
+    make kiwikeebsmacro:default:flash
```
Comment (fauxpark) [inline keyboards/kiwikeebs/macro_v2/readme.md:17]:
```suggestion
    make kiwikeebs/macro_v2:default

Flashing example for this keyboard:

    make kiwikeebs/macro_v2:default:flash
```

File: keyboards/kiwikeebs/readme.md [hunk 9]
```diff
@@ -1,19 +1,19 @@
 # KiwiKeebs MacroBoard V1.0
```
Comment (fauxpark) [inline keyboards/kiwikeebs/readme.md:1]: This file should be deleted.

File: keyboards/kiwikeebs/macro_v2/readme.md [hunk 10]
```diff
@@ -0,0 +1,19 @@
+# KiwiKeebs MacroBoard V1.0
```
Comment (fauxpark) [inline keyboards/kiwikeebs/macro_v2/readme.md:1]:
```suggestion
# KiwiKeebs MacroBoard V2.0
```

File: keyboards/kiwikeebs/macro_v2/macro.h [hunk 11]
```diff
@@ -0,0 +1,35 @@
+/* Copyright 2020 KiwiKeebs
```
Comment (fauxpark) [inline keyboards/kiwikeebs/macro_v2/macro.h:1]: This file should be deleted.

File: keyboards/kiwikeebs/macro_v2/macro2.0.h [hunk 12]
```diff
@@ -0,0 +1,35 @@
+/* Copyright 2020 KiwiKeebs
```
Comment (fauxpark) [inline keyboards/kiwikeebs/macro_v2/macro2.0.h:1]: This file should be renamed to `macro_v2.h`.

File: keyboards/kiwikeebs/macro_v2/macro.c [hunk 13]
```diff
@@ -0,0 +1,29 @@
+/* Copyright 2020 KiwiKeebs
```
Comment (fauxpark) [inline keyboards/kiwikeebs/macro_v2/macro.c:1]: This file should be deleted.

File: keyboards/kiwikeebs/macro_v2/macro2.0.c [hunk 14]
```diff
@@ -0,0 +1,29 @@
+/* Copyright 2020 KiwiKeebs
```
Comment (fauxpark) [inline keyboards/kiwikeebs/macro_v2/macro2.0.c:1]: This file should be renamed to `macro_v2.c`.

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/kiwikeebs/macro_v2/info.json [hunk 15]
```diff
@@ -0,0 +1,21 @@
+{
+    "keyboard_name": "KiwiKeebs MacroBoard v.10",
```
[the remark this answers is not in the document]
Comment (AKiwi92) [reply keyboards/kiwikeebs/macro_v2/info.json]: Done!

**Comment by AKiwi92** [comment]
@fauxpark thanks for taking the time to look over it, appreciate it!

Did all your changes manually just so I could look for myself and keep an eye for anything else :)

File: keyboards/kiwikeebs/macro_v2/macro_v2.c [hunk 16]
```diff
@@ -0,0 +1,29 @@
...
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "macro2.0.h"
```
Comment (fauxpark) [inline keyboards/kiwikeebs/macro_v2/macro_v2.c:17]:
```suggestion
#include "macro_v2.h"
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Comment by AKiwi92** [comment]
Totally missed this change, done :)

File: keyboards/kiwikeebs/macro/keymaps/via/keymap.c [hunk 17]
```diff
@@ -0,0 +1,34 @@
...
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+    /* Base */
+    [_BASE] = LAYOUT(
+        KC_DEL,  KC_HOME, LT(_FN,KC_END),
+        KC_MPRV, KC_MPLY, KC_MNXT, KC_MUTE
+    ),
+    [_FN] = LAYOUT(
+        RESET,    KC_UP,   _______,
+        KC_LEFT, KC_DOWN, KC_RGHT, MAGIC_TOGGLE_NKRO
+    )
```
Comment (fauxpark) [inline keyboards/kiwikeebs/macro/keymaps/via/keymap.c:33]: Need two more empty layers here.

File: keyboards/kiwikeebs/macro_v2/keymaps/via/keymap.c [hunk 18]
```diff
@@ -0,0 +1,34 @@
...
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+    /* Base */
+    [_BASE] = LAYOUT(
+        KC_DEL,  KC_HOME, LT(_FN,KC_END),
+        KC_MPRV, KC_MPLY, KC_MNXT, KC_MUTE
+    ),
+    [_FN] = LAYOUT(
+        RESET,    KC_UP,   _______,
+        KC_LEFT, KC_DOWN, KC_RGHT, MAGIC_TOGGLE_NKRO
+    )
```
Comment (fauxpark) [inline keyboards/kiwikeebs/macro_v2/keymaps/via/keymap.c:33]: Need two more empty layers here.

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/kiwikeebs/macro/keymaps/via/keymap.c (hunk 17 above)
Comment (AKiwi92) [reply keyboards/kiwikeebs/macro/keymaps/via/keymap.c:33]: Added!

File: keyboards/kiwikeebs/macro_v2/keymaps/via/keymap.c (hunk 18 above)
Comment (AKiwi92) [reply keyboards/kiwikeebs/macro_v2/keymaps/via/keymap.c:33]: Added!

**Comment by AKiwi92** [comment]
@fauxpark Let me know if there is anything else you need me to do! :)

File: keyboards/kiwikeebs/macro_v2/info.json [hunk 19]
```diff
@@ -0,0 +1,21 @@
+{
+    "keyboard_name": "KiwiKeebs MacroBoard v.2.0",
+    "url": "https://www.reddit.com/r/mechmarket/comments/ibijrd/ic_kiwikeebs_6key_macroboard_rotary_encoder_diy/",
+    "maintainer": "KiwiKeebs",
+    "width": 5,
+    "height": 2,
```
Comment (fauxpark) [inline keyboards/kiwikeebs/macro_v2/info.json:6]:
```suggestion
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Review by fauxpark — APPROVED** [review]

**Comment by AKiwi92** [comment]
@fauxpark thanks so much for all the help it is much appreciated!

Is there anything else I need to do now this has been approved? Or is it just a case of waiting a while?

Thanks again!

**Comment by AKiwi92** [comment]
@fauxpark @drashna Hi guys sorry I am still pretty new to GitHub is there anything else I need to do or is it just a case of waiting from this point? I just don't want to hold anything up from my end!

Cheers :)

**Comment by drashna** [comment]
Just a case of waiting, sorry. 

And thanks!

**Pull request merged**